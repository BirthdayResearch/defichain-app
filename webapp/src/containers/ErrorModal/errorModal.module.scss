// App and Bootstrap variables required
@import "~bootstrap/scss/functions";
@import "~bootstrap/scss/variables";
@import "~bootstrap/scss/mixins";
@import "../../scss/variables";
@import "../../scss/bootstrap/variables";

.dark {
  background-color: $gray-900 !important;
  color: $white;
}

.errorModal {
  text-align: center;
  margin-top: 6rem;

  .errorModalContent {
    display: inline-block;
  }

  svg {
    color: $dfi-red;
  }
}

.onContentModal {
  height: 400px;
  width: 400px !important;
}

.openErrorModal {
  filter: blur(5px);
}