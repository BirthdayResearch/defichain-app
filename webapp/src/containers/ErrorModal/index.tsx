import React from 'react';
import { connect } from 'react-redux';
import { I18n } from 'react-redux-i18n';
import { RiErrorWarningLine, RiLoader4Line } from 'react-icons/ri';
import Loader from '../../components/Loader';
import styles from './errorModal.module.scss';

interface ErrorModalProps {
  isRestart: boolean;
  showWarning: boolean;
}

const ErrorModal: React.FunctionComponent<ErrorModalProps> = (
  props: ErrorModalProps
) => (
  <>
    <div className={styles.errorModal}>
      {!props.isRestart && props.showWarning && (
        <div className={styles.errorModalContent}>
          <RiErrorWarningLine size={100} />
          <p>{I18n.t('alerts.nodeDisconnected')}</p>
        </div>
      )}
      {props.isRestart && (
        <div>
          <Loader
            className='mb-5'
            size={100}
            color={'#ff0000'}
            borderSize={10}
          />
          <p>{I18n.t('alerts.restartNode')}</p>
        </div>
      )}
    </div>
  </>
);
const mapStateToProps = (state) => {
  const { isRestart, showWarning } = state.errorModal;
  return {
    isRestart,
    showWarning,
  };
};

export default connect(mapStateToProps)(ErrorModal);
