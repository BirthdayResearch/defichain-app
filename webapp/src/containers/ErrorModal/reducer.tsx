import { createSlice } from '@reduxjs/toolkit';

const configSlice = createSlice({
  name: 'errorModal',
  initialState: {
    isOpen: false,
    isRestart: false,
    showWarning: false,
  },
  reducers: {
    openErrorModal(state) {
      state.isOpen = true;
      state.showWarning = true;
    },
    closeErrorModal(state) {
      state.isOpen = false;
      state.isRestart = false;
      state.showWarning = false;
    },
    restartModal(state) {
      state.isRestart = true;
      state.isOpen = true;
    },
  },
});

const { actions, reducer } = configSlice;

export const { openErrorModal, closeErrorModal, restartModal } = actions;

export default reducer;
