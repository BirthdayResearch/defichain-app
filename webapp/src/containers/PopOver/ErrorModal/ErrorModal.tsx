import React from 'react';
import { connect } from 'react-redux';
import { I18n } from 'react-redux-i18n';
import { RiErrorWarningLine } from 'react-icons/ri';
import Loader from '../../../components/Loader';
import styles from '../popOver.module.scss';

interface ErrorModalProps {
  isRestart: boolean;
  showWarning: boolean;
}

const ErrorModal: React.FunctionComponent<ErrorModalProps> = (
  props: ErrorModalProps
) => (
  <>
    <div className={styles.errorModal}>
      {!props.isRestart && props.showWarning && (
        <div className={styles.errorModalContent}>
          <RiErrorWarningLine size={50} />
          <p>{I18n.t('alerts.nodeDisconnected')}</p>
        </div>
      )}
      {props.isRestart && (
        <div>
          <Loader className='mb-4' size={28} />
          <p>{I18n.t('alerts.restartNode')}</p>
        </div>
      )}
    </div>
  </>
);

const mapStateToProps = (state) => {
  const { isRestart, showWarning } = state.popover;
  return {
    isRestart,
    showWarning,
  };
};

export default connect(mapStateToProps)(ErrorModal);
