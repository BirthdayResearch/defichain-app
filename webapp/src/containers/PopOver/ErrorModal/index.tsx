import React from 'react';
import ErrorComponent from './ErrorModal';
import { connect } from 'react-redux';
import { Modal, ModalBody } from 'reactstrap';
import styles from '../popOver.module.scss';

interface ErrorModalProps {
  isOpen: boolean;
}

function ErrorModal(props: ErrorModalProps) {
  return (
    <Modal
      isOpen={props.isOpen}
      centered
      contentClassName={styles.onContentModal}
    >
      <ModalBody>
        <ErrorComponent />
      </ModalBody>
    </Modal>
  );
}

const mapStateToProps = ({ popover }) => ({
  isOpen: popover.isOpen,
});

export default connect(mapStateToProps)(ErrorModal);
