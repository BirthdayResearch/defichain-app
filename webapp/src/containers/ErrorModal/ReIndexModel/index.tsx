import React from 'react';
import { connect } from 'react-redux';
import { I18n } from 'react-redux-i18n';
import { Button, ModalHeader } from 'reactstrap';
import { Modal, ModalBody, ModalFooter } from 'reactstrap';
import { MdClose } from 'react-icons/md';
import { restartNodeWithReIndexing, closeApp } from '../../../utils/isElectron';
import { closeReIndexModal, isRestartLoader } from '../reducer';

interface ReIndexModalProps {
  isReIndexModelOpen: boolean;
  isReIndexRestart: boolean;
  closeReIndexModal: () => void;
  isRestartLoader: () => void;
}

const ReIndexModal: React.FunctionComponent<ReIndexModalProps> = (
  props: ReIndexModalProps
) => {
  const restartAppWithReIndexing = () => {
    props.closeReIndexModal();
    props.isRestartLoader();
    restartNodeWithReIndexing({ isReindexReq: true });
  };

  const closePopupAndApp = () => {
    props.closeReIndexModal();
    closeApp();
  };

  return (
    <div>
      <Modal isOpen={props.isReIndexModelOpen} centered>
        <ModalHeader>
          <MdClose onClick={props.closeReIndexModal} />
        </ModalHeader>
        <ModalBody>
          <p className='text-center'>
            {I18n.t('alerts.restartAppWithReindexNoticeline1')}
          </p>
          <p className='text-center'>
            {I18n.t('alerts.restartAppWithReindexNoticeline2')}
          </p>
        </ModalBody>
        <ModalFooter>
          <div className='d-flex justify-content-end'>
            <Button color='primary' onClick={restartAppWithReIndexing}>
              {I18n.t('alerts.yesRestartAppWithReindex')}
            </Button>
            <Button className='ml-4' onClick={closePopupAndApp}>
              {I18n.t('alerts.noCloseApp')}
            </Button>
          </div>
        </ModalFooter>
      </Modal>
    </div>
  );
};

const mapStateToProps = (state) => {
  const { isReIndexModelOpen, isReIndexRestart } = state.errorModal;

  return {
    isReIndexModelOpen,
    isReIndexRestart,
  };
};

const mapDispatchToProps = {
  closeReIndexModal,
  isRestartLoader,
};

export default connect(mapStateToProps, mapDispatchToProps)(ReIndexModal);
