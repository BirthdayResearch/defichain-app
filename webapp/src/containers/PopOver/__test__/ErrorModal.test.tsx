import React from 'react';
import { Provider } from 'react-redux';
import store from '../../../app/rootStore';
import Enzyme from '../../../utils/testUtils/enzyme';
import ErrorModal from '../ErrorModal';

describe('Error Modal', () => {
  it('should check for snapshot', () => {
    const wrapper = Enzyme.mount(
      <Provider store={store}>
        <ErrorModal />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
