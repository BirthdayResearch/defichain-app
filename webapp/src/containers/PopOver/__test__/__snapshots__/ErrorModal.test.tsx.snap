// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Modal should check for snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(ErrorModal)>
    <ErrorModal
      dispatch={[Function]}
      isOpen={false}
    >
      <Modal
        autoFocus={true}
        backdrop={true}
        backdropTransition={
          Object {
            "mountOnEnter": true,
            "timeout": 150,
          }
        }
        centered={true}
        container="body"
        fade={true}
        isOpen={false}
        keyboard={true}
        modalTransition={
          Object {
            "timeout": 300,
          }
        }
        onClosed={[Function]}
        onOpened={[Function]}
        returnFocusAfterClose={true}
        role="dialog"
        scrollable={false}
        trapFocus={false}
        unmountOnClose={true}
        zIndex={1050}
      />
    </ErrorModal>
  </Connect(ErrorModal)>
</Provider>
`;
